udp: add UV_UDP_MMSG_FREE recv_cb flag (#2836)
libuv/libuv: docs/src/udp.rst

    When using :man:`recvmmsg(2)`, chunks will have the `UV_UDP_MMSG_CHUNK` flag set,
    those must not be freed. There will be a final callback with `nread` set to 0,
    `addr` set to NULL and the buffer pointing at the initially allocated data with
    the `UV_UDP_MMSG_CHUNK` flag cleared and the `UV_UDP_MMSG_FREE` flag set.

>     This is a good chance for the callee to free the provided buffer.

Reviewer: ```suggestion
    The callee can now safely free the provided buffer.
```
Or something along those lines? "A good chance" sounds somewhat inconclusive.
